app/models/petition.rb

```diff
@@ -28,6 +28,7 @@ class Petition < ActiveRecord::Base
 
   has_perishable_token called: 'sponsor_token'
 
+  before_save :stamp_parliament_response_at, if: -> { response.present? && response_was.nil? }
   after_create :set_petition_on_creator_signature
```

Comment:
Oh, sorry for the duplication here. I wrote my comment before @otlaitil responded - just forgot to hit "go".